dependencies.yaml

```
@@ -755,7 +755,12 @@ dependencies:
           - ipython
           - openpyxl
   notebook_cuda_version:
-    common:
-      - output_types: conda
-        packages:
-          - cuda-version=12.0
+    specific:
+       - output_types: conda
+         matrices:
+           - matrix: {cuda: "12.*"}
+             packages:
+               - cuda-version=12.0
+           - matrix: {cuda: "11.*"}
+             packages:
+               - cuda-version=11.8
```

We probably want to match `major.minor` exactly. If we're only testing on CUDA 12.0 and 11.8, then we shouldn't use `12.*`. Otherwise we'll get conflicts if the environment was generated for a different CUDA minor version.

It's my intention to file a follow-up to #14644 that separates `cuda-version` declaration and CUDA package declaration . The `cuda-version` should always match the requested `major.minor` version. The CUDA package definitions only depend on the major version, like `cuda: "12.*"` matrix installing `cuda-nvcc` and other packages from the 12.x CTK, while `cuda: "11.*"` would install `cudatoolkit`.
```suggestion
           - matrix: {cuda: "12.0"}
             packages:
               - cuda-version=12.0
           - matrix: {cuda: "11.8"}
             packages:
               - cuda-version=11.8
```